js/hexaworld.js

```diff
@@ -0,0 +1,236 @@
+var hexaworld = {
+
+  $schema: "http://json-schema.org/schema#",
+  id: "https://github.com/hexaworld/hexaworld-schema/schema.js",
+
+  definitions: {
+
+    // basic definitions
+
+    pixelCoord: {
+      type: "array",
+      minItems: 2,
+      maxItems: 3,
+      items: { type: "number"}
+    },
+
+    axialCoord: {
+      type: "array",
+      minItems: 2,
+      maxItems: 2,
+      items: { type: "integer" }
+    },
+
+    color: {
+      oneOf: [
+        { type: "object",
+          properties: {
+            hue: {
+              type: "number",
+              minimum: 0,
+              maximum: 360
+            },
+            saturation: {
+              type: "number",
+              minimum: 0,
+              maximum: 100
+            },
+            value: {
+              type: "number",
+              minimum: 0,
+              maximum: 100
+            }
+          }
+        },
+        { type: "object",
+          properties: {
+            r: {
+              type: "number",
+              minimum: 0,
+              maximum: 255
+            },
+            g: {
+              type: "number",
+              minimum: 0,
+              maximum: 255
+            },
+            b: {
+              type: "number",
+              minimum: 0,
+              maximum: 255
+            }
+          }
+        }
+      ]
+    },
+
+    //  geometric definitions
+
+    scaleFactor: {
+      type: "number",
+      minimum: 0,
+      maximum: 1
+    },
+
+    rotationDegrees: {
+      type: "number",
+      minimum: -180,
+      maximum: 180
+    },
+
+    transformation: {
+      scale: {
+        type: "object",
+        properties: {
+          x: { $ref: "#/definitions/scaleFactor" },
+          y: { $ref: "#/definitions/scaleFactor" },
+          z: { $ref: "#/definitions/scaleFactor" },
+        }
+      },
+      rotation: {
+        type: "object",
+        properties: {
+          x: { $ref: "#/definitions/rotationDegrees" },
+          y: { $ref: "#/definitions/rotationDegrees" },
+          z: { $ref: "#/definitions/rotationDegrees" },
+        }
+      }
+    },
+
+    shape: {
+      type: "object",
+      properties: {
+        color: { $ref: "#/definitions/color" },
+        transformation: { $ref: "#/definitions/transformation" },
+
+        // TODO: how should we represent textures?
+        texture: { type: "string" }
+      }
+    },
+
+    ellipsoid: {
+      allOf: [
+        { $ref: "#/definitions/shape" },
+        { properties: {
+            center: { $ref: "#/definitions/pixelCoord" },
+            a: { type: "number" },
+            b: { type: "number" },
+            c: { type: "number" }
+          }
+        }
+      ]
+    },
+
+    rectangle: {
```

Comment:
Just as an example because I imagine it'll be the most commonly used polygon. Think if anything we should add more specific polygons, because it'll make it easier to see what kind of shape's there are in a tile.